parsimonious/grammar.py

```diff
@@ -296,64 +298,74 @@ def __init__(self, custom_rules=None):
         """
         self.custom_rules = custom_rules or {}
 
-    def visit_parenthesized(self, parenthesized, (left_paren, _1,
-                                                  expression,
-                                                  right_paren, _2)):
+    def visit_parenthesized(self, parenthesized, _a):
```

Comment:
Yeah, I agree. It's made a lot of things more verbose, especially sorting a list of tuples according to a custom key. It's one of the few changes I'd consider a straight-up regression from Python 2 to 3.  Still, Python 3.5 now has enough new features and enough library support that it's unequivocally the choice for new projects, and the "six" approach to developing has generally won over the 2to3 approach from what I've seen in libraries trying to support both versions.

As regards the naming, the `Visitor` API passes two arguments, which in my visitors I usually name `def visit_FOO(self, node, children):` 

I almost never care about `node` _unless_ I want `node.text`, otherwise, I unpack (and possibly deserialize) elements of the tuple of children. What do you think about changing the naming convention in this file to something like the following?
```python
def visit_parenthesized(self, node, parenthesized):
    """Treat a parenthesized subexpression as just its contents.

    Its position in the tree suffices to maintain its grouping semantics.

    """
    (left_paren, _1, expression, right_paren, _2) = parenthesized
    return expression
```